File: 剑指offer/05. 替换空格.py
class Solution:
    def replaceSpace(self, s: str) -> str:
        list_ = []
        rep = "%20"
        for i in s:
            if i != " ":
                list_.append(i)
            else:
                list_.append(rep)
        return ''.join(list_)            #### .join()函数实现list->str
    
时间复杂度 O(N)： 遍历使用 O(N) ，每轮添加（修改）字符操作使用 O(1)；
空间复杂度 O(N)： Python 新建的 list 和 Java 新建的 StringBuilder 都使用了线性大小的额外空间。

https://leetcode-cn.com/problems/ti-huan-kong-ge-lcof/solution/mian-shi-ti-05-ti-huan-kong-ge-ji-jian-qing-xi-tu-/

        

File: 剑指offer/C++_Version/05. 替换空格.cpp
//方法1
class Solution {
public:
    string replaceSpace(string s) {
        int count=0, len = s.size();
        for(char c: s){
            if(c==' ') 
                count ++;
        }
        s.resize(len + 2*count);
        for(int i=len-1,j=s.size()-1; i<j; i--,j--){
            if(s[i]!=' ')
                s[j] = s[i];
            else{
                s[j-2]='%';
                s[j-1]='2';
                s[j]='0';
                j -= 2;
            }
        }
        return s;
    }   
};
